framework/JobHandler.py

```diff
@@ -439,6 +441,34 @@ def isThisJobFinished(self, identifier):
     ## problems.
     self.raiseAnError(RuntimeError,"Job "+identifier+" is unknown!")
 
+  def areTheseJobsFinished(self, uniqueHandler="any"):
+    """
+      Method to check if all the runs in the queue are finished
+      @ In, uniqueHandler, string, optional, it is a special keyword attached to
+        each runner. If provided, just the jobs that have the uniqueIdentifier
+        will be retrieved. By default uniqueHandler = 'any' => all the jobs for
+        which no uniqueIdentifier has been set up are going to be retrieved
+      @ Out, isFinished, bool, True all the runs in the queue are finished
+    """
+    uniqueHandler = uniqueHandler.strip()
+    with self.__queueLock:
+      for run in self.__finished:
+        if run.uniqueHandler == uniqueHandler:
+          return False
+
+      for queue in [self.__queue, self.__clientQueue]:
+        for run in queue:
+          if run.uniqueHandler == uniqueHandler:
+            return False
+
+      for run in self.__running + self.__clientRunning:
+        if run is not None and run.uniqueHandler == uniqueHandler:
+          return False
+
+    self.raiseADebug("The jobs with uniqueHandler ", uniqueHandler, "are finished")
+
+    return True
+
```

Comment:
To make maintenance more feasible, do you think there is a way that we can combine `isThisJobFinished` and `areTheseJobsFinished`? Or are uniqueHandler and identifier too different? Yeah, maybe this is not feasible. At first glance, I thought this function just took a group of identifiers, but uniqueHandler is different from identifier.

Reply:
As a side note not related to this PR, we may want to rework these names so they are not so synonymous.

Reply:
For my understanding, identifier and uniqueHandler are different. Identifier is a unique ID for the each job, while uniqueHandler is a label to identify a group of jobs.

Reply:
Yeah, I caught myself here, but wanted to preserve the comment because of the potential confusion later on given the similarity of these method names.  I think your description gives a big hint as to what we should be renaming the `uniqueHandler`: `groupIdentifier`, `groupId` or `groupID`. The term "handler" makes me think this is an object of some sort, and if it was meant to be "handle" then I would say it conflicts with "identifier" since those terms don't have enough distinction to differentiate these.

Again, forgive my rant as these are beyond the scope of this PR, but I wanted to voice my concern before I forgot :P

Reply:
Good point, I agree with you.